lib/rules/no-dom-import.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const { getDocsUrl } = require('../utils');
+
+const DOM_TESTING_LIBRARY_MODULES = [
+  'dom-testing-library',
+  '@testing-library/dom',
+];
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow importing from DOM Testing Library',
+      category: 'Best Practices',
+      recommended: false,
+      url: getDocsUrl('no-dom-import'),
+    },
+    messages: {
+      noDomImport:
+        'import from DOM Testing Library is restricted, import from corresponding Testing library framework instead',
```

Comment:
The error message sounds unclear to me. What do you think?

Reply:
Yeah, I'm not convinced at all about it but I can't suggest which module should be imported. What I can do is to check which DTL module is used (`dom-testing-library` or `@testing-library/dom`) and say something like `import directly from '@testing-library/dom' is restricted, instead you must import from '@testing-library/<framework>'` by literally leaving that "<framework>" placeholder

Reply:
Yes, I would think of something like that too. I see two solutions possible to that: 

1. For each rule, we could configure it by adding the selected framework and then use it in the proper rule. For example: 

```json
{
  "rules": {
    "testing-library/no-dom-import": ["error", "react"]
  }
}
```

However, this would require the user to add the framework to each rule where we need to differentiate the frameworks.

2. Generate a set of rules for every corresponding framework in this [file](https://github.com/Belco90/eslint-plugin-testing-library/blob/master/lib/index.js). This could become:

```js
module.exports = {
  rules,
  configs: {
    recommended: {
      plugins: ['testing-library'],
      rules: recommendedRules,
    },
    react: {
      plugins: ['testing-library'],
      rules: { ...generateRecommendedRules("react"), 'testing-library/no-debug': 'warn' },
    },
    // ...
  },
}
```

And in `generateRecommendedRules`:

```js
function generateRecommendedRules(framework) {
  return allRules.map(rule => {
    const augmentedRule = rule(framework)
    return augmentedRule
  })
}
```

Where `rule` would be a higher-order function that returns the proper rule configuration based on the framework: 

```js
'use strict';

const { getDocsUrl } = require('../utils');

const DOM_TESTING_LIBRARY_MODULES = [
  'dom-testing-library',
  '@testing-library/dom',
];

module.exports = (framework) => {
  return {
    meta: {
      type: 'problem',
      docs: {
        description: 'Disallow importing from DOM Testing Library',
        // ...
     },
    messages: {
      noDomImport:
        `import from DOM Testing Library is restricted, import from "@testing-library/${framework}`,
    },
    fixable: null,
    schema: [],
  },
  // ...
}
```

What do you think?

Reply:
🤦‍♂️ of course, didn't think about context options! Really interesting approach, I'd go for something intermediate between those 2 options: we add the framework option to this rule through [context.options](https://eslint.org/docs/developer-guide/working-with-rules#contextoptions) (so you receive it within `create` function through context arg and pass it to report message, as this seems the standard instead of doing the whole workaround to pass it to the rule function itself) and then we setup automatically this option on each recommended framework config (so it's automatically setup for those frameworks but everyone can customize it when adding the rule manually).

So I'm gonna merge this one (as I wanted to improve the error message but obviously it's gonna be way better when the rule options are received so I'm gonna skip that for now) and create an issue for you to handle that separately, but only for receiving the context.options and improve the message with proper framework setup, nothing yet about fixable. Are you happy with that?

Reply:
Brilliant!